### PR TITLE
Add Fint score

### PR DESCRIPTION
**Pull request by JenkeScheen**
## Description
Compute the Fitness Interaction Score (FINTscore) given a dict with interactions generated by PLIP. This is a fast-and-dirty method of computing a score based on interactions while taking Fitness data into account.

## Todos

- [x] Refactor usage of PLIP to standalone component
- [x] Add FINTscore so it can call PLIP directly
- [x] Add unit tests for FINTscore
- [x] Add FINTscore to docking workflows

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT license as defined in our [LICENSE](https://github.com/choderalab/asapdiscovery/blob/main/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).

### CODE REVIEW

**Comment by JenkeScheen** [comment]
@hmacdope ready for you to hook up to docking workflows, see https://github.com/choderalab/asapdiscovery/pull/749/commits/1ae3d5432e9845ca8cf7ae6c370ba73733da29d9 for usage example.

File: asapdiscovery-data/asapdiscovery/data/plip.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    penalty_multipliers = 1
+    reward_multipliers = 1
+    intn_score_bucket = []
+
+    # iterate over each interaction that was found.
+    intn_dict = get_interactions_plip(protein, pose, "fitness", target)
+
+    for _, data in intn_dict.items():
+        # if the interaction is with backbone, add a reward to the score.
+        if data["color"] == "#008000":
```
Comment (hmacdope) [inline asapdiscovery-data/asapdiscovery/data/plip.py:314]: Using the colour here is fragile as it is not strongly coupled with the code that generates the colours if you know what I mean. Is there another easy way to do this? If not happy to let slide for now as long as we make a note of it.

Comment (JenkeScheen) [reply asapdiscovery-data/asapdiscovery/data/plip.py:323]: Agreed, it also makes the `metadata` parameter yaml more opaque. I didn't see an obvious way of doing it more logically while I was building the MVP so I just took this route - will try to find a better one.

**Comment by hmacdope** [comment]
Confirmed this work for a Postera upload. Merging!

File: asapdiscovery-data/asapdiscovery/data/plip.py (hunk 1 above)
Comment (hmacdope) [reply asapdiscovery-data/asapdiscovery/data/plip.py:324]: Raised #785 to address this, but first lets get this MVP shipped and we can work on refining it.

**Review by hmacdope — APPROVED** [review]
Amazing work @JenkeScheen! You got this build in record time.

**Pull request merged**